README.md

```diff
@@ -241,53 +1038,845 @@ Track 404 errors in Google Analytics by adding this to your `notFound.html` temp
+
+**Advanced Approach (When you need more features):**
+- Use relationship fields: `_author`, `_featuredImage`
+- Link to user profiles, image management, etc.
+- Full power of ApostropheCMS relationships
+
+**The module automatically detects which format you're using** and generates correct structured data either way.
+
+### Featured Images
+
+Several schema types rely on a `_featuredImage` relationship field being present on your document.
+
+**Schema types that use featured images:**
+- **Product** - Product image
+- **Recipe** - Recipe photo
+- **How To** - Guide illustration
+- **Video Object** - Video thumbnail (falls back to featured image)
+
+**Example implementation:**
+```javascript
+// modules/article/index.js
+export default {
+  extend: '@apostrophecms/piece-type',
+  options: {
+    label: 'Article'
+  },
+  fields: {
+    add: {
+      _featuredImage: {
+        label: 'Featured Image',
+        type: 'relationship',
+        withType: '@apostrophecms/image',
+        max: 1,
+        required: true  // Make required if using Product or Recipe schemas
+      },
+    },
+    group: {
+      basics: {
+        fields: ['title', '_featuredImage']
+      }
+    }
+  }
+};
+```
+
+**Note:** The field name **must be** `_featuredImage` (with the leading underscore) for the SEO module to find it automatically.
+
+### Paywalled Content
+
+If you mark content as paywalled, your templates must use consistent CSS classes or IDs to wrap premium content. The module needs to know which HTML element contains the paywalled content.
+
+**How it works:**
+
+1. Add a wrapper element around your paywalled content in your template
+2. Configure the CSS selector in the SEO settings to match your wrapper
+
+**Example template implementation:**
+
+```nunjucks
+{# views/show.html #}
+<article>
+  <h1>{{ data.piece.title }}</h1>
+  
+  {# Free preview content #}
+  <div class="article-preview">
+    {{ data.piece.excerpt }}
+  </div>
+  
+  {# Paywalled content - note the class name #}
+  <div class="paywall">
+    {% if data.user %}
+      {# Show full content to subscribers #}
+      {{ data.piece.body }}
+    {% else %}
+      {# Show paywall message to non-subscribers #}
+      <div class="paywall-notice">
+        <p>Subscribe to read more...</p>
+      </div>
+    {% endif %}
+  </div>
+</article>
+```
+
+**Common CSS selector patterns:**
+
+```css
+/* By class (most common) */
+.paywall
+.premium-content
+.members-only
+
+/* By ID */
+#paywalled-content
+
+/* By data attribute */
+[data-paywall="true"]
+
+/* Multiple classes */
+.article-body.premium
+```
+
+**In the SEO settings**, set the "Paywall CSS Selector" field to match your implementation (e.g., `.paywall`).
+
+**Why this matters:** Google requires you to explicitly mark which parts of your page require payment. The CSS selector tells search engines exactly where the paywall boundary is, helping them show appropriate content previews without penalties.
```

Comment:
Any special notes for Astro? I guess they just have to make sure they do this in Astro.